vendor/pischem/caddy/src/etc/inc/plugins.inc.d/caddy.inc

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * Copyright (C) 2023 Cedrik Pischem
+ * Copyright (C) 2017 EURO-LOG AG
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+function caddy_services()
+{
+    global $config;
+
+    $services = array();
+
+    if (isset($config['Pischem']['caddy']['general']['enabled']) &&
+        $config['Pischem']['caddy']['general']['enabled'] == 1) {
+        $services[] = array(
+            'description' => gettext('Caddy Web Server'),
+            'configd' => array(
+                'restart' => array('caddy restart'),
+                'start' => array('caddy start'),
+                'stop' => array('caddy stop'),
+            ),
+            'name' => 'caddy',
+            'pidfile' => '/var/run/caddy.pid'
+        );
+    }
+
+    return $services;
+}
+
+function caddy_xmlrpc_sync()
+{
+    $result = array();
+
+    $result[] = array(
+        'description' => gettext('Caddy Web Server'),
+        'section' => 'Pischem.caddy',
```

Comment:
I think we have to straighten out this path for upstream inclusion.

Reply:
Do you mean the section has to become "OPNsense.caddy" ?
I don't understand why the xmlrpc sync works on my HA firewalls then with this section name.

Reply:
Since this works for my HA setup and I guess it revolves around the Namespace I think we can leave it this way for now?